ConcurrencyInGo/primeFinder/main.go

```diff
@@ -133,23 +134,33 @@ func main() {
 
 	}
 
-	done0 := make(chan interface{})
-	aLotOfInts := make([]int, 5)
-	for i := 5; i < 1000000; i++ {
+	const getNumberOfPrimes = 3000000
+
+	aLotOfInts := make([]int, 50000)
+	for i := 5; i < getNumberOfPrimes; i++ {
 		aLotOfInts = append(aLotOfInts, i)
 	}
 
+	done0 := make(chan interface{})
+	defer close(done0)
+
 	primeStream := generator(done0, aLotOfInts...)
-	//	primeStream := generator(done0, 1, 2, 3, 4, 5, 1729, 17, 47)
-	numFinders := 10
+
+	numFinders := runtime.NumCPU()
 	finders := make([]<-chan int, numFinders)
 	for i := 0; i < numFinders; i++ {
 		finders[i] = notSmartPrime1(done0, primeStream)
 	}
 
-	for p := range take(done0, fanIn(done0, finders...), 100) {
```

Comment:
For some reason I need to add the if p == nil to stop the range when using the take function.